src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp

```diff
@@ -0,0 +1,257 @@
+}
+
+void itkProcessRegistrationOptimusToolBox::run(void)
+{
+    medRegistrationSelectorToolBox *toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+    if (toolbox)
+    {
+        if (toolbox->process() && (toolbox->process()->identifier() == "itkProcessRegistrationOptimus"))
+        {
+            d->process = toolbox->process();
+
+        }
+        else
+        {
+            d->process = new itkProcessRegistrationOptimus;
+            toolbox->setProcess(d->process);
+        }
+        medAbstractData *fixedData = toolbox->fixedData();
+        medAbstractData *movingData = toolbox->movingData();
+
+        if (!fixedData || !movingData)
+            return;
+
+        // Many choices here
+
+        itkProcessRegistrationOptimus *process_Registration = dynamic_cast<itkProcessRegistrationOptimus *>(d->process);
+        if (!process_Registration)
+        {
+            qWarning() << "registration process doesn't exist";
+            return;
+        }
+        process_Registration->setNumberOfHistogramBins(d->binsBox->value());
+        process_Registration->setNumberOfInterpolations(d->interpolationsBox->value());
+        process_Registration->setNumberOfIterations(d->iterationsBox->value());
+        process_Registration->setNumberOfSpatialSamples(d->samplesBox->value());
+        process_Registration->setRhoEnd(d->rhoEndBox->value());
+        process_Registration->setRhoStart(d->rhoStartBox->value());
+        process_Registration->setScalingCoefficient(d->scalingCoeffBox->value());
+
+
+        // process->setMyWonderfullParameter(fronTheGui);
+        // process->setMyWonderfullParameter(fronTheGui);
+
+        // or ...
+
+        // itkProcessRegistrationOptimus *process = new itkProcessRegistrationOptimus;
+        // process->setMyWonderfullParameter(fronTheGui);
+        // process->setMyWonderfullParameter(fronTheGui);
+
+        // ----
+
+        d->process->setFixedInput(fixedData);
+        d->process->setMovingInput(movingData);
+
+        medRunnableProcess *runProcess = new medRunnableProcess;
+        runProcess->setProcess(d->process);
+
+        d->progression_stack->addJobItem(runProcess, tr("Progress:"));
+        d->progression_stack->setActive(runProcess, true);
+        d->progression_stack->disableCancel(runProcess);
+        connect(runProcess, SIGNAL(success(QObject*)), this, SIGNAL(success()));
+        connect(runProcess, SIGNAL(failure(QObject*)), this, SIGNAL(failure()));
+        connect(runProcess, SIGNAL(cancelled(QObject*)), this, SIGNAL(failure()));
+        //First have the moving progress bar,
+        //and then display the remaining % when known
+        connect(runProcess, SIGNAL(activate(QObject*, bool)),
+            d->progression_stack, SLOT(setActive(QObject*, bool)));
+
+        medJobManagerL::instance()->registerJobItem(runProcess, d->process->identifier());
+        QThreadPool::globalInstance()->start(dynamic_cast<QRunnable*>(runProcess));
```

Comment:
replace these paragraphs by `this->addConnectionsAndStartJob(runProcess);` and if needed `enableOnProcessSuccessImportOutput(runProcess, false);`